test/e2e/util.go

```diff
@@ -787,10 +787,14 @@ func testContainerOutputInNamespace(scenarioName string, c *client.Client, pod *
 		Failf("Failed to create pod: %v", err)
 	}
 
-	containerName := pod.Spec.Containers[0].Name
-
 	// Wait for client pod to complete.
-	expectNoError(waitForPodSuccessInNamespace(c, pod.Name, containerName, ns))
+	var containerName string
+	for id, container := range pod.Spec.Containers {
+		expectNoError(waitForPodSuccessInNamespace(c, pod.Name, container.Name, ns))
```

Comment:
I dont think so. 
From where i stand, all of the containers should be running for test cases involved multiple ones. Only some of the output needs to be verified though.